gems/aws-sdk-core/lib/aws-sdk-core/shared_config.rb

```diff
@@ -176,61 +165,49 @@ def credentials_present?
 
     def assume_role_from_profile(cfg, profile, opts, chain_config)
       if cfg && prof_cfg = cfg[profile]
-        opts[:source_profile] ||= prof_cfg["source_profile"]
+        opts[:source_profile] ||= prof_cfg['source_profile']
         credential_source = opts.delete(:credential_source)
-        credential_source ||= prof_cfg["credential_source"]
+        credential_source ||= prof_cfg['credential_source']
         if opts[:source_profile] && credential_source
-          raise Errors::CredentialSourceConflictError.new(
-            "Profile #{profile} has a source_profile, and "\
-              "a credential_source. For assume role credentials, must "\
-              "provide only source_profile or credential_source, not both."
-          )
+          raise Errors::CredentialSourceConflictError, "Profile #{profile} has a source_profile, and "\
```

Comment:
For errors, it seems most of the previous code liked to initialize errors but it can just raise the class instead. I see it got changed here. Going forward, what do you think about consistently adding the message onto the next line?

```
raise Errors::CredentialSourceConflictError,
      "my very long"\
      "message"
```

Reply:
Yeah - I agree with that.  General convention should be raise the Class and put long error messages starting on a new line.